core/base/compactTriangulation/CompactTriangulation.h

```diff
@@ -0,0 +1,4133 @@
+/// \ingroup base
+/// \class ttk::CompactTriangulation
+/// \author Guoxi Liu <guoxil@g.clemson.edu>
+/// \date November 2021.
+///
+/// \brief CompactTriangulation is a class implemented based on the
+/// TopoCluster data structure, which is a localized data structure for
+/// simplicial meshes.
+/// The key idea of TopoCluster is to subdivide the simplicial mesh into
+/// clusters. Then, the connectivity information is computed locally for each
+/// cluster and discarded when it is no longer needed. The simplicial mesh needs
+/// to be subdivided before using TopoCluster, i.e., there has to be a scalar
+/// field named "ttkCompactTriangulationIndex" to denote the cluster index of
+/// each vertex. Note Topocluster will reindex the simplices based on the
+/// clustering input array.
+/// \b Related \b publications \n
+/// "TopoCluster: A Localized Data Structure for Topology-based Visualization"
+/// Guoxi Liu, Federico Iuricich, Riccardo Fellegara, and Leila De Floriani
+/// IEEE Transactions on Visualization and Computer Graphics, 2021.
+///
+/// \sa ttk::Triangulation
+/// \sa ttk::CompactTriangulationPreconditioning
+
+#pragma once
+
+// base code includes
+#include <AbstractTriangulation.h>
+#include <CellArray.h>
+#include <FlatJaggedArray.h>
+#include <algorithm>
+#include <boost/functional/hash.hpp>
+#include <boost/unordered_map.hpp>
+#include <boost/unordered_set.hpp>
+#include <list>
+
+#define EDGE_ID 1
+#define TRIANGLE_ID 2
```

Comment:
I would favor a private (if possible) `enum class` inside `ImplicitCluster` to reduce the scope of these `define`.